src/test/java/com/google/sps/data/DeliverySystemTest.java

```diff
@@ -0,0 +1,167 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.maps.errors.ApiException;
+import java.io.IOException;
+import java.lang.InterruptedException;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests that orders are correctly assigned to delivery slots.
+ */
+@RunWith(JUnit4.class)
+public class DeliverySystemTest {
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private ArrayList<Point> points;
+  private List<CourierStop> courierStops;
+  private Point startPoint;
+  private DeliverySystem deliverySystem;
+  private OrderHandler orderHandler;
+  private ManhattanDistancePathFinder pathFinder;
+
+  @Before
+  public void setUp() throws ApiException, BadRequestException, DataNotFoundException, IOException, InterruptedException {
+    helper.setUp();
+    pathFinder = new ManhattanDistancePathFinder();
+    deliverySystem = new DeliverySystem(pathFinder);
+    orderHandler = new OrderHandler(pathFinder);
+
+    points = new ArrayList();
+    points.add(new LibraryPoint(0.0002, 0.0002, 0));
+    points.add(new Point(0.0003, 0.0003));
+    points.add(new LibraryPoint(0.0004, 0.0005, 1));
+    points.add(new Point(0.0005, 0.0006));
+    points.add(new Point(0.0009, 0.0008));
+    startPoint = new Point(-0.0001, 0);
+    courierStops = Arrays.asList(startPoint, points.get(0), points.get(1), points.get(2), 
+      points.get(3), points.get(4))
+      .stream()
+      .map(point -> new CourierStop((Point)point))
+      .collect(Collectors.toList());
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  private void createOrder(CourierStop library, CourierStop recipient, List<String> bookIds, String userId, boolean addOrderKey) {
+    String orderKey = orderHandler.addOrderToDatastore((LibraryPoint)library.getPoint(), bookIds, userId, recipient.getPoint());
+    if (addOrderKey) {
+      library.addOrderKey(orderKey);
+      recipient.addOrderKey(orderKey);
+    }
+  }
+
+  @Test
+  public void testTakeAllOrders() throws ApiException, BadRequestException, DataNotFoundException, EntityNotFoundException, IOException, InterruptedException  { 
```

Comment:
nit: try to work out the conditions in the test names, to make it clear what the test is testing:
TakeAllOrdersWhereSlotIsLargeEnough
TakeNoOrderIfSlotTooSmall
TakeOnlyOneOrderIfSecondOrderExceedsDeliverySlot

Reply:
Done.